tests/framework/http_api.py

```diff
@@ -58,15 +58,25 @@ def __init__(self, api, resource, id_field=None):
     def get(self):
         """Make a GET request"""
         url = self._api.endpoint + self.resource
-        res = self._api.session.get(url)
+        try:
+            res = self._api.session.get(url)
+        except Exception as e:
+            if self._api.error_callback:
+                self._api.error_callback("GET", self.resource, str(e))
+            raise
```

Comment:
You could avoid all the duplication by delegating to `request`:
```
res = self.request("GET", url)
```

This was not done initially because it saved nothing, but now you can save the error handling duplication.

Reply:
So I wanted to do that at first, too, but request() and get() have different error conditions. request() asserts that we get 204 NO CONTENT (unless we have an error response), while get() asserts we get 200 OK with a json body. So it's not that straightforward I think?

Reply:
Oh right I did not see that. You could move that check to request and then dispatch on the method `if GET and not 200, else ...`

But I will leave that up to you.